src/FluxRM.jl

```diff
@@ -17,19 +19,54 @@ end
 
 mutable struct Flux
     handle::Ptr{API.flux_t}
+    fd::RawFD
 
-    function Flux(handle::Ptr{API.flux_t})
+    function Flux(handle::Ptr{API.flux_t}; own=true)
         @assert handle != C_NULL
-        this = new(handle)
-        finalizer(this) do flux
-            API.flux_close(flux)
+        fd = API.flux_pollfd(handle)
+        systemerror("flux_pollfd", fd < 0)
+
+        this = new(handle, RawFD(fd))
+        # alternative unique through WeakRefDict?
+        if own
+            finalizer(this) do flux
+                API.flux_close(flux)
+            end
         end
         return this
     end
 end
 Base.unsafe_convert(::Type{Ptr{API.flux_t}}, flux::Flux) = flux.handle
 # flux_fatal_set
 
+function progress(reactor)
+    while true
```

Comment:
In conversation we decided that it should be sufficient to run `API.flux_reactor_run(reactor, API.FLUX_REACTOR_NOWAIT)` once. `poll_fd` only get's the connection progress so there are other reactors that would need to make progress.